t/rose-metadata-check/12-template-variables.t

```diff
@@ -0,0 +1,56 @@
+#!/usr/bin/env bash
+#-------------------------------------------------------------------------------
+# Copyright (C) British Crown (Met Office) & Contributors.
+#
+# This file is part of Rose, a framework for meteorological suites.
+#
+# Rose is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# Rose is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with Rose. If not, see <http://www.gnu.org/licenses/>.
+#-------------------------------------------------------------------------------
+# Test "rose metadata-check":
+# Check that hack allowing section name [template variables] works.
+# Follows example in https://github.com/metomi/rose/issues/2737
+
+. $(dirname $0)/test_header
+
+CHECKS=(
+    "template variables=FOO == 1",
+    "any(template variables=FOO == 1)"
+    "len(template variables=FOO) < 42"
+    "template variables=FOO) in [2, 4, 8]"
+    "len(template variables=FOO) < 42 and any(template variables=FOO == 1)"
+    )
+
+
+# 3 tests for each case:
+tests $(( ${#CHECKS[@]} * 3 ))
+
+setup
+
+# Allows array members to have spaces without splitting:
+for ((i = 0; i < ${#CHECKS[@]}; i++)); do
+    CHECK="${CHECKS[$i]}"
+
+    TEST_KEY="$TEST_KEY_BASE::${CHECK}"
+    init <<__META_CONFIG__
+[template variables=FOO]
+fail-if=${CHECK}
+__META_CONFIG__
+
+    run_pass "$TEST_KEY" rose metadata-check -C ../config
```

Comment:
Can you try adding a `rose macro -V` test to this to make sure this fix works for other commands too.